tests/UI/campaigns/functional/BO/02_orders/01_orders/viewAndEditOrder/11_checkInvoice.js

```diff
@@ -0,0 +1,1229 @@
+
+      const textResult = await taxesPage.enableEcoTax(page, true);
+      await expect(textResult).to.be.equal('Update successful');
+    });
+  });
+
+  // Pre-condition - Create 4 products
+  [virtualProduct,
+    customizedProduct,
+    productWithSpecificPrice,
+    productWithEcoTax,
+  ].forEach((product, index) => {
+    describe(`PRE-TEST: Create product '${product.name}'`, async () => {
+      if (index === 0) {
+        it('should go to \'Catalog > Products\' page', async function () {
+          await testContext.addContextItem(this, 'testIdentifier', 'goToProductsPage', baseContext);
+
+          await dashboardPage.goToSubMenu(page, dashboardPage.catalogParentLink, dashboardPage.productsLink);
+
+          await productsPage.closeSfToolBar(page);
+
+          const pageTitle = await productsPage.getPageTitle(page);
+          await expect(pageTitle).to.contains(productsPage.pageTitle);
+        });
+
+        it('should reset all filters and get number of products', async function () {
+          await testContext.addContextItem(this, 'testIdentifier', 'resetFiltersBeforeCreate', baseContext);
+
+          numberOfProducts = await productsPage.resetAndGetNumberOfLines(page);
+          await expect(numberOfProducts).to.be.above(0);
+        });
+      }
+
+      it('should go to add product page', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `goToAddProductPage${index}`, baseContext);
+
+        if (index === 0) {
+          await productsPage.goToAddProductPage(page);
+        } else {
+          await addProductPage.goToAddProductPage(page);
+        }
+
+        const pageTitle = await addProductPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(addProductPage.pageTitle);
+      });
+
+      it('should create Product', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `createProduct${index}`, baseContext);
+
+        const createProductMessage = await addProductPage.createEditBasicProduct(page, product);
+
+        if (product === customizedProduct) {
+          await addProductPage.addCustomization(page, product.customization);
+        }
+
+        if (product === productWithSpecificPrice) {
+          await addProductPage.addSpecificPrices(page, product.specificPrice);
+        }
+        if (product === productWithEcoTax) {
+          await addProductPage.addEcoTax(page, product.ecoTax);
+        }
+        await expect(createProductMessage).to.equal(addProductPage.settingUpdatedMessage);
+      });
+    });
+  });
+
+  // Pre-condition - Create second order from FO
+  createOrderSpecificProductTest(secondOrderByCustomer, baseContext);
+
+  // Check invoice
+  [
+    {
+      args: {
+        orderRow: 2,
+        deliveryAddress: Address.second,
+        billingAddress: Address.third,
+        product: virtualProduct,
+        productQuantity: 13,
+        tax: `${virtualProduct.tax} %`,
```

Comment:
Could you create this combinations before the test and explain what each one mean please ?

Reply:
These combinations are not for creation, on each one we check an invoice with different data.
I will add some comments.

Reply:
We should know what we test here to be able to reproduce it manually.